tests/unit/test_skiplist_graph.c

```diff
@@ -0,0 +1,198 @@
+  Node *cur_node;
+  SIValue *cur_prop;
+
+  char *keys[] = {"5.5", "0", "-30.2", "7", "1", "2", "-1.5", NULL};
+
+  // The IDs we will assign to the Node values in the skiplist
+  // (defined as the order the keys should be in after sorting)
+  long ids[] = {5, 2, 0, 6, 3, 4, 1};
+
+  for (long i = 0; keys[i] != NULL; i ++) {
+    cur_node = NewNode(ids[i], node_label);
+    cur_prop = malloc(sizeof(SIValue));
+    SIValue_FromString(cur_prop, keys[i]);
+    Node_Add_Properties(cur_node, 1, &prop_key, cur_prop);
+    skiplistInsert(sl, cur_prop, cur_node);
+  }
+
+  long last_id = 3;
+  skiplistIterator iter;
+  // Iterate over a range of keys [1, INF)
+  SIValue min = SI_DoubleVal(1);
+  iter = skiplistIterateRange(sl, &min, NULL, 1, 0);
+  while ((cur_node = skiplistIterator_Next(&iter)) != NULL) {
+    assert(last_id + 1 == cur_node->id);
+    last_id = cur_node->id;
+  }
+
+  skiplistFree(sl);
+}
+
+void test_skiplist_delete(void) {
+  int delete_result;
+  SIValue prop_to_delete;
+
+  skiplist *sl = build_skiplist();
+
+  SIValue_FromString(&prop_to_delete, words[2]);
+  skiplistNode *old_skiplist_node = skiplistFind(sl, &prop_to_delete);
+  Node *node_to_delete = old_skiplist_node->vals[0];
+  SIValue_FromString(&prop_to_delete, words[3]);
+
+  // Attempt to delete a non-existent key-value pair
+  delete_result = skiplistDelete(sl, &prop_to_delete, node_to_delete);
+  assert(delete_result == 0);
+
+  // Attempt to delete a non-existent skiplist key
+  SIValue_FromString(&prop_to_delete, "not_a_key");
+  delete_result = skiplistDelete(sl, &prop_to_delete, NULL);
+  assert(delete_result == 0);
+
+  // Delete a single value from the skiplist
+  delete_result = skiplistDelete(sl, Node_Get_Property(node_to_delete, prop_key), node_to_delete);
+  assert(delete_result == 1);
+
+  // Delete full nodes from the skiplist - a single key and all values that share it
+  SIValue_FromString(&prop_to_delete, words[3]);
+  delete_result = skiplistDelete(sl, &prop_to_delete, NULL);
+  assert(delete_result == 1);
+
+  // Verify that the skiplistNode has been deleted
+  void *search_result = skiplistFind(sl, &prop_to_delete);
+  assert(search_result == NULL);
+
+  skiplistFree(sl);
+}
+
+void test_skiplist_update(void) {
+  skiplistNode *search_result;
+  skiplist *sl = build_skiplist();
+  SIValue find_prop;
+  SIValue_FromString(&find_prop, words[2]);
+  skiplistNode *old_skiplist_node = skiplistFind(sl, &find_prop);
+  Node *node_to_update = old_skiplist_node->vals[0];
+
+  SIValue *new_prop = malloc(sizeof(SIValue));
+  SIValue old_prop = SI_Clone(*Node_Get_Property(node_to_update, prop_key));
+  SIValue_FromString(new_prop, "updated_val");
+  // Update an existing key-value pair in the index
+  skiplistNode *new_skiplist_node = update_skiplist(sl, node_to_update, &old_prop, new_prop);
+  // Update an existing key-value pair in the Node properties
+  GraphEntity_Update_Property((GraphEntity *)node_to_update, prop_key, new_prop);
```

Comment:
It's more likely that we start by updating node's property and as a result reindex.